GDJS/GDJS/IDE/ExporterHelper.cpp

```diff
@@ -808,6 +811,45 @@ void ExporterHelper::ExportObjectAndBehaviorsIncludes(
   }
 }
 
+void ExporterHelper::ExportObjectAndBehaviorsRequiredFiles(
+    const gd::Project &project, std::vector<gd::String> &requiredFiles) {
+  auto addRequiredFiles = [&](const std::vector<gd::String> &newRequiredFiles) {
+    for (const auto &requiredFile : newRequiredFiles) {
+      InsertUnique(requiredFiles, requiredFile);
+    }
+  };
+
+  auto addObjectRequiredFiles = [&](const gd::Object &object) {
+    // Ensure needed files are included for the object type and its behaviors.
+
+    // TODO Do the same for objects once we have a case to test it.
+    // No object use WebAssembly library so the requiredFiles attribute
+    // is not implemented yet.
+
+    std::vector<gd::String> behaviors = object.GetAllBehaviorNames();
+    for (std::size_t j = 0; j < behaviors.size(); ++j) {
+      const gd::BehaviorMetadata &metadata =
+          gd::MetadataProvider::GetBehaviorMetadata(
+              JsPlatform::Get(),
+              object.GetBehavior(behaviors[j]).GetTypeName());
+      addRequiredFiles(metadata.requiredFiles);
+    }
+  };
+
+  auto addObjectsRequiredFiles =
+      [&](const gd::ObjectsContainer &objectsContainer) {
+        for (std::size_t i = 0; i < objectsContainer.GetObjectsCount(); ++i) {
+          addObjectRequiredFiles(objectsContainer.GetObject(i));
+        }
+      };
+
+  addObjectsRequiredFiles(project);
+  for (std::size_t i = 0; i < project.GetLayoutsCount(); ++i) {
+    const gd::Layout &layout = project.GetLayout(i);
+    addObjectsRequiredFiles(layout);
+  }
```

Comment:
Not "super happy" to have this "logic" traversing objects duplicated, I wonder if it's worth making it into a function/template, or not?

In the future, we might have objects in other places (like... inside objects? :))

Reply:
Might be too much forward thinking and not useful though.

Reply:
I like the Iterator pattern 😃

Reply:
But, yes I think it can wait.

Reply:
Totally.